packages/serverless/src/awslambda.ts

```diff
@@ -274,17 +274,20 @@ export function wrapHandler<TEvent, TResult>(
       }, timeoutWarningDelay) as unknown as NodeJS.Timeout;
     }
 
-    // Applying `sentry-trace` to context
-    let traceparentData;
-    const eventWithHeaders = event as { headers?: { [key: string]: string } };
-    if (eventWithHeaders.headers && isString(eventWithHeaders.headers['sentry-trace'])) {
-      traceparentData = extractTraceparentData(eventWithHeaders.headers['sentry-trace']);
-    }
+    const hub = getCurrentHub();
 
-    const baggageHeader = eventWithHeaders.headers && eventWithHeaders.headers.baggage;
-    const dynamicSamplingContext = baggageHeaderToDynamicSamplingContext(baggageHeader);
+    const eventWithHeaders = event as { headers?: { [key: string]: string } };
 
-    const hub = getCurrentHub();
+    const sentryTrace =
+      eventWithHeaders.headers && isString(eventWithHeaders.headers['sentry-trace'])
+        ? eventWithHeaders.headers['sentry-trace']
+        : undefined;
```

Comment:
```suggestion
    const sentryTrace = eventWithHeaders.headers?.['sentry-trace'] || undefined;
```

as we are already using optional chaining, might as well use it here as well ;)

Reply:
We need the `isString` check - see https://github.com/getsentry/sentry-javascript/pull/8425#discussion_r1247949623 for more details.